fitbenchmarking/controllers/aoclda_controller.py

```diff
@@ -0,0 +1,327 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+############################################################
+
+"""
+Implements a controller for AMD AOCL Data Analytics NLLS solver
+https://github.com/amd/aocl-data-analytics
+"""
+
+import traceback
+
+import numpy as np
+from aoclda.nonlinear_model import nlls
+
+from fitbenchmarking.controllers.base_controller import Controller
+from fitbenchmarking.utils.exceptions import UnknownMinimizerError
+
+
+class AOCLDAController(Controller):
+    """
+    Controller for the AOCLDA NLLS fitting software.
+    """
+
+    algorithm_check = {
+        "all": [
+            "gn",
+            "hybrid",
+            "newton",
+            "newton-tensor",
+            "gn_reg",
+            "hybrid_reg",
+            "newton_reg",
+            "newton-tensor_reg",
+        ],
+        "ls": [
+            "gn",
+            "hybrid",
+            "newton",
+            "newton-tensor",
+            "gn_reg",
+            "hybrid_reg",
+            "newton_reg",
+            "newton-tensor_reg",
+        ],
+        "deriv_free": [],
+        "general": [],
+        "simplex": [],
+        "trust_region": ["gn", "hybrid", "newton", "newton-tensor"],
+        "levenberg-marquardt": ["gn", "gn_reg"],
+        "gauss_newton": ["gn", "gn_reg"],
+        "bfgs": [],
+        "conjugate_gradient": [],
+        "steepest_descent": [],
+        "global_optimization": [],
+        "MCMC": [],
+    }
+
+    jacobian_enabled_solvers = [
+        "gn",
+        "hybrid",
+        "newton",
+        "newton-tensor",
+        "gn_reg",
+        "hybrid_reg",
+        "newton_reg",
+        "newton-tensor_reg",
+    ]
+
+    hessian_enabled_solvers = [
+        "hybrid",
+        "newton",
+        "newton-tensor",
+        "hybrid_reg",
+        "newton_reg",
+        "newton-tensor_reg",
+    ]
+
+    support_for_bounds = True
+
+    def __init__(self, cost_func):
+        """
+        Initialises variable used for temporary storage.
+
+        :param cost_func: Cost function object selected from options.
+        :type cost_func: subclass of
+                :class:`~fitbenchmarking.cost_func.base_cost_func.CostFunc`
+        """
+        super().__init__(cost_func)
+
+        self.VERBOSE = 0
+
+        # AOCLDA data
+        self._handle = None
+        self._lb = None
+        self._ub = None
+        self._x = None
+        self._status = None
+
+        # All these options should be passed
+        self.maxit = 500  # Same as RALFit
+        self.ftol = 1e-5
+        self.abs_ftol = 1e-6  # Similar to Ceres?
+        self.gtol = 1e-8
+        self.abs_gtol = 1e-10  # Similar to Ceres?
+        self.xtol = 1e-8
+
+        # These are passed as a new jacobian type
+        use_fd = False
+        # fd_step = 1e-7
+        # fd_ttol = 1e-4
+
+        # call-backs
+        self.aoclda_jac = (
+            self.jac_eval if self.cost_func.jacobian and not use_fd else None
```

Comment:
As mentioned in #1492, I think we should be adding in a new finite difference Jacobian option in a separate PR, and remove the `use_fd` flag here